### PR TITLE
For staging environment, include dynamic text for `dandi download` command

### LINKED ISSUE #1553: For empty dandiset listing upload instructions, make them use instance URL not the main one
image

lists `dandi download https://dandiarchive.org/dandiset/207699/draft` although it is for https://gui-staging.dandiarchive.org/dandiset/207699/draft/files?location= .  Information on the instance url is known AFAIK so should be used there instead of hardcoded.

### PR DESCRIPTION
**Pull request by aaronkanzer**
When using `gui-staging.dandiarchive.org`, a minor inconsistency was observed where the CLI command was incorrect.

This PR makes a simple fix to render the appropriate text depending on if the user is using staging or production.

Closes #1553 

Cc @kabilar

### CODE REVIEW

**Review by kabilar — COMMENTED** [review]
Thank you @aaronkanzer.  This looks good to me.

Hi @waxlamp, when you have a chance, this pull request is ready for review.  I am not a JavaScript developer so it would be good to have you review.  Thank you.

File: web/src/components/FileBrowser/FileUploadInstructions.vue [hunk 1]
```diff
@@ -53,4 +53,16 @@ import { useDandisetStore } from '@/stores/dandiset';
...
+
+const downloadCommand = computed(() => {
+  const baseUrl = import.meta.env.VITE_APP_DANDI_API_ROOT === 'https://api-staging.dandiarchive.org/api/'
+    ? 'https://gui-staging.dandiarchive.org/dandiset/'
+    : 'https://dandiarchive.org/dandiset/';
+
+  return dandisetIdentifier.value
+    ? `> dandi download ${baseUrl}${dandisetIdentifier.value}/draft`
+    : ''; // Empty string just as a fallback in case store.dandiset? is undefined
+});
```
Comment (yarikoptic) [inline web/src/components/FileBrowser/FileUploadInstructions.vue:65]: this is a good first step since would allow both main and staging one but it is not a generic solution.  

The dandi-cli is actually "too smart" to support a wide range of URLs and they do not have to be associated with a specific "short named" instance (if something fails -- it is a bug):

```
❯ dandi download --help
Usage: dandi download [OPTIONS] [URL]...

  Download files or entire folders from DANDI.

  Accepted resource identifier patterns:
   - DANDI:<dandiset id>[/<version>]
   - https://gui.dandiarchive.org/...
   - https://identifiers.org/DANDI:<dandiset id>[/<version id>] (<version id> cannot be 'draft')
   - https://<server>[/api]/[#/]dandiset/<dandiset id>[/<version>][/files[?location=<path>]]
   - https://*dandiarchive-org.netflify.app/...
   - https://<server>[/api]/dandisets/<dandiset id>[/versions[/<version>]]
   - https://<server>[/api]/assets/<asset id>[/download]
   - https://<server>[/api]/dandisets/<dandiset id>/versions/<version>/assets/<asset id>[/download]
   - https://<server>[/api]/dandisets/<dandiset id>/versions/<version>/assets/?path=<path>
   - https://<server>[/api]/dandisets/<dandiset id>/versions/<version>/assets/?glob=<glob>
   - dandi://<instance name>/<dandiset id>[@<version>][/<path>]
   - https://<server>/...
```

so here IMHO it just should use local server URL... chatgpt suggested some code like this

```vue
computed: {
  baseUrl() {
    const protocol = window.location.protocol;
    const host = window.location.hostname;
    const port = window.location.port ? `:${window.location.port}` : '';
    return `${protocol}//${host}${port}`;
  }
}
```

Comment (aaronkanzer) [reply web/src/components/FileBrowser/FileUploadInstructions.vue:65]: To clarify though -- for example, a user on `identifiers.org` though does not see the DANDI CLI download command, so I'm curious why not confine it just to the two specific places where it might render.

I do agree that it _could_ be more generic; however, I'd vote to keep it simple and confined in case an unknown edge case exists (e.g. `windoww.location.protocol` is tied to the browser, not Vue, etc., etc.)

**Comment by waxlamp** [comment]
Just a note to @aaronkanzer and @kabilar: thanks for contributing this; I'm taking a look.

File: web/src/components/FileBrowser/FileUploadInstructions.vue (hunk 1 above)
Comment (waxlamp) [reply web/src/components/FileBrowser/FileUploadInstructions.vue:63]: I pushed a few commits to clean up the PR and change the URL generation to something close to Yarik's ChatGPT code. Check it out and let me know what you think.

Comment (yarikoptic) [reply web/src/components/FileBrowser/FileUploadInstructions.vue:63]: > To clarify though -- for example, a user on `identifiers.org` though does not see the DANDI CLI download command, so I'm curious why not confine it just to the two specific places where it might render.

sorry, missed the question -- what "two specific places" you have in mind?

here the list of supported URLs organically grown to make it easier for a user to download, e.g. "just cut paste URL from the browser". identifiers.org is indeed niche use case, and would work only for the main instance unless we add some other identifiers to handle in addition to `DANDI`.  Indeed it is not really needed and likely not really used, but if `dandi` is asked to download it and it knows how to parse it -- why not? ;)

File: web/src/views/DandisetLandingView/DownloadDialog.vue [hunk 2]
```diff
@@ -122,7 +122,7 @@ import CopyText from '@/components/CopyText.vue';
 
 function formatDownloadCommand(identifier: string, version: string): string {
   if (version === 'draft') {
-    return `dandi download https://dandiarchive.org/dandiset/${identifier}/draft`;
+    return `dandi download ${window.location.origin}/dandiset/${identifier}/draft`;
```
[the remark this answers is not in the document]
Comment (yarikoptic) [reply web/src/views/DandisetLandingView/DownloadDialog.vue:125]: @aaronkanzer do you think you would have time to add custom handling below for `DANDI:` as well? it is pretty much "use DANDI: if window.location.origin == 'https://dandiarchive.org'", and otherwise needs a `${window.location.origin}/dandiset/` prefix

[the remark this answers is not in the document]
Comment (aaronkanzer) [reply web/src/views/DandisetLandingView/DownloadDialog.vue:125]: Updated here -- let me know if this is what you had in mind -- thanks again

[the remark this answers is not in the document]
Comment (yarikoptic) [reply web/src/views/DandisetLandingView/DownloadDialog.vue:125]: I think it looks good. 

Should/could we proceed here or any other aspect remains open?

**Comment by yarikoptic** [comment]
@aaronkanzer ping on above -- could it be considered done??

**Comment by aaronkanzer** [comment]
@yarikoptic works for me -- if you want to approve, I'm happy to merge.

File: web/src/components/FileBrowser/FileUploadInstructions.vue [hunk 3]
```diff
@@ -53,4 +53,12 @@ import { useDandisetStore } from '@/stores/dandiset';
 
 const store = useDandisetStore();
 const dandisetIdentifier = computed(() => store.dandiset?.dandiset.identifier);
+
+if (dandisetIdentifier.value === undefined) {
+  throw new Error('store.dandiset must be defined');
+}
+
+const downloadCommand = computed(() => {
+  return `dandi download ${window.location.origin}/dandiset/${dandisetIdentifier.value}/draft`
```
Comment (waxlamp) [inline web/src/components/FileBrowser/FileUploadInstructions.vue:62]: I did not really follow the discussion in the [other thread](https://github.com/dandi/dandi-archive/pull/1810/files#r1466534015). Could you summarize here whether this logic is correct for the upload instructions? Is there a reason not to use the same logic as in the download dialog below? Specifically, if a user downloads their dandiset via `dandi download DANDI:123456/draft`, will the subsequent `dandi upload` command get confused about where to upload?

File: web/src/views/DandisetLandingView/DownloadDialog.vue [hunk 4]
```diff
@@ -120,14 +120,16 @@ import { computed, ref } from 'vue';
...
+function downloadCommand(identifier: string, version: string): string {
+  // Use the special 'DANDI:' url prefix if appropriate.
+  const generalUrl = `${window.location.origin}/dandiset/${identifier}`;
+  const dandiUrl = `DANDI:${identifier}`;
+  const url = window.location.origin == 'https://dandiarchive.org' ? dandiUrl : generalUrl;
+
+  // Prepare a url suffix to specify a specific version (or not).
+  const versionPath = version ? `/${version}` : '';
+
+  return `dandi download ${url}${versionPath}`;
```
Comment (waxlamp) [inline web/src/views/DandisetLandingView/DownloadDialog.vue:132]: I have simplified the code and I think it still covers all cases. Please let me know if you disagree.

**Review by waxlamp — COMMENTED** [review]
@yarikoptic (cc: @aaronkanzer), I think this is ready to go but I had a couple of questions (see the inline comments below).

You can see this in action for a non-dandiarchive URL at the [Netlify deploy preview](https://deploy-preview-1810--gui-staging-dandiarchive-org.netlify.app/dandiset/214613). We will need to deploy to production to see it in action on the canonical dandiarchve.org instance, but it's low risk to do that since it won't disrupt the actual functioning of the archive and we can deploy quick fixes if anything is wrong.

Let me know what you think.

**Comment by aaronkanzer** [comment]
> @yarikoptic (cc: @aaronkanzer), I think this is ready to go but I had a couple of questions (see the inline comments below).
> 
> You can see this in action for a non-dandiarchive URL at the [Netlify deploy preview](https://deploy-preview-1810--gui-staging-dandiarchive-org.netlify.app/dandiset/214613). We will need to deploy to production to see it in action on the canonical dandiarchve.org instance, but it's low risk to do that since it won't disrupt the actual functioning of the archive and we can deploy quick fixes if anything is wrong.
> 
> Let me know what you think.

This sounds good to me, as it won't disturb the API via testing in a prod UI environment -- thanks for the updates here

File: web/src/views/DandisetLandingView/DownloadDialog.vue (hunk 4 above)
Comment (aaronkanzer) [reply web/src/views/DandisetLandingView/DownloadDialog.vue:132]: This makes sense to me -- thanks Roni.

File: web/src/components/FileBrowser/FileUploadInstructions.vue (hunk 3 above)
Comment (aaronkanzer) [reply web/src/components/FileBrowser/FileUploadInstructions.vue:62]: The DANDI CLI _should_ still be OK -- as @yarikoptic mentioned that any short-hand prefix shouldn't break here, as "The dandi-cli is actually "too smart" to support a wide range of URLs and they do not have to be associated with a specific "short named" instance (if something fails -- it is a bug)"

I can test further if you'd like more confirmation

**Comment by waxlamp** [comment]
> > @yarikoptic (cc: @aaronkanzer), I think this is ready to go but I had a couple of questions (see the inline comments below).
> > You can see this in action for a non-dandiarchive URL at the [Netlify deploy preview](https://deploy-preview-1810--gui-staging-dandiarchive-org.netlify.app/dandiset/214613). We will need to deploy to production to see it in action on the canonical dandiarchve.org instance, but it's low risk to do that since it won't disrupt the actual functioning of the archive and we can deploy quick fixes if anything is wrong.
> > Let me know what you think.
> 
> This sounds good to me, as it won't disturb the API via testing in a prod UI environment -- thanks for the updates here

Great, then I'll go ahead and merge this and we can still have a discussion about the questions in my self-review above (and do followup PRs in case anything needs to be fixed). Thanks!

**Review by waxlamp — APPROVED** [review]

**Comment by yarikoptic** [comment]
> You can see this in action for a non-dandiarchive URL at the [Netlify deploy preview](https://deploy-preview-1810--gui-staging-dandiarchive-org.netlify.app/dandiset/214613). We will need to deploy to production to see it in action on the canonical dandiarchve.org instance, but it's low risk to do that since it won't disrupt the actual functioning of the archive and we can deploy quick fixes if anything is wrong.
> 
> Let me know what you think.

sounds sensible -- let's just try to keep this in memory so we could check as soon as release comes out.  Any plans to release soon?

**Comment by waxlamp** [comment]
Seems to be working nicely 😸
